pkg/cmd/cli/cmd/set/probe.go

```diff
@@ -0,0 +1,409 @@
+	}
+	count := 0
+	if o.Command != nil {
+		count++
+	}
+	if len(o.OpenTCPSocket) > 0 {
+		count++
+	}
+	if len(o.HTTPGet) > 0 {
+		count++
+	}
+
+	switch {
+	case o.Remove && count != 0:
+		return fmt.Errorf("--remove may not be used with any flag except --readiness or --liveness")
+	case count > 1:
+		return fmt.Errorf("you may only set one of --get-url, --open-tcp, or command")
+	case len(o.OpenTCPSocket) > 0 && intOrString(o.OpenTCPSocket).IntVal > 65535:
+		return fmt.Errorf("--open-tcp must be a port number between 1 and 65535 or an IANA port name")
+	}
+	if o.FailureThreshold != nil && *o.FailureThreshold < 1 {
+		return fmt.Errorf("--failure-threshold may not be less than one")
+	}
+	if o.SuccessThreshold != nil && *o.SuccessThreshold < 1 {
+		return fmt.Errorf("--success-threshold may not be less than one")
+	}
+	if o.InitialDelaySeconds != nil && *o.InitialDelaySeconds < 0 {
+		return fmt.Errorf("--initial-delay-seconds may not be negative")
+	}
+	if o.TimeoutSeconds != nil && *o.TimeoutSeconds < 0 {
+		return fmt.Errorf("--timeout-seconds may not be negative")
+	}
+	if o.PeriodSeconds != nil && *o.PeriodSeconds < 0 {
+		return fmt.Errorf("--period-seconds may not be negative")
+	}
+	return nil
+}
+
+func (o *ProbeOptions) Run() error {
+	infos := o.Infos
+	singular := len(o.Infos) <= 1
+	if o.Builder != nil {
+		loaded, err := o.Builder.Do().IntoSingular(&singular).Infos()
+		if err != nil {
+			return err
+		}
+		infos = loaded
+	}
+
+	patches := CalculatePatches(infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		transformed := false
+		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *kapi.PodSpec) error {
+			containers, _ := selectContainers(spec.Containers, o.ContainerSelector)
+			if len(containers) == 0 {
+				fmt.Fprintf(o.Err, "warning: %s/%s does not have any containers matching %q\n", info.Mapping.Resource, info.Name, o.ContainerSelector)
+				return nil
+			}
+			// perform updates
+			transformed = true
+			for _, container := range containers {
+				o.updateContainer(container)
+			}
+			return nil
+		})
+		return transformed, err
+	})
+	if singular && len(patches) == 0 {
+		return fmt.Errorf("%s/%s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
+	}
+
+	if o.PrintObject != nil {
+		var infos []*resource.Info
+		for _, patch := range patches {
+			info := patch.Info
+			if patch.Err != nil {
+				fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+				continue
+			}
+			infos = append(infos, info)
+		}
+		object, err := resource.AsVersionedObject(infos, !singular, "", nil)
+		if err != nil {
+			return err
+		}
+		return o.PrintObject(object)
+	}
+
+	failed := false
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+			continue
+		}
+
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
```

Comment:
I remember trying to add a similar check re empty patch in `oc env` but @liggitt was against it.